pkg/sdk/agent.go

```diff
@@ -186,3 +188,38 @@ func generateMetadata(userID string, privateKey crypto.PrivateKey) (metadata.MD,
 	kv[auth.SignatureMetadataKey] = base64.StdEncoding.EncodeToString(signature)
 	return metadata.New(kv), nil
 }
+
+func (sdk *agentSDK) IMAMeasurements(ctx context.Context, privKey any, resultFile *os.File) ([]byte, error) {
+	request := &agent.IMAMeasurementsRequest{}
+
+	md, err := generateMetadata(string(auth.ConsumerRole), privKey)
```

Comment:
thid consumer role is part of the computation manifest so it should not be included here